### PR TITLE
Add encoder for stream transports (TCP/TLS).

### LINKED ISSUE #16: Add packet encode/decode support for stream protocols (TCP, TCP over TLS).
There is a two byte length field that proceeds the message when sent over stream protocols like TCP as defined in RFC 1035. I have a patch for this I will submit as a pull request.

### PR DESCRIPTION
**Pull request by pusateri**
Fixes #16.
I'll add example TLS test code for completeness in the future but the payload is the same for TCP.

### CODE REVIEW

**Comment by silverwind** [comment]
Can you add a few words to README about these methods and why they're useful?

File: index.js [hunk 1]
```diff
@@ -743,6 +743,29 @@ exports.encodingLength = function (result) {
...
+  exports.streamEncode.bytes = len + 2
+  return buf
+}
+
+exports.streamEncode.bytes = 0
+
+exports.streamDecode = function (buf) {
+  const len = buf.readUInt16BE(0)
+  if (buf.length < len) {
+    return {}
```
Comment (mafintosh) [inline index.js]: in case we end up wanting to land this, this should return `null` or throw in this case

**Comment by pusateri** [comment]
Addressed review comments.

**Comment by silverwind** [comment]
Am I understanding this correctly that this mode could be used for https://github.com/mafintosh/dns-socket/issues/11?

**Comment by pusateri** [comment]
Yes, retrying over TCP is the traditional way to solve truncated responses. This pull request adds the encoding for TCP.

**Comment by silverwind** [comment]
I don't quite get the purpose of `exports.stream{En,De}code.bytes`, what are they for?

**Comment by pusateri** [comment]
I was just mimicking exports.encode.bytes and exports.decode.bytes. I'm not sure what they are for either. I figured if existing applications are using these, they may want the stream version. But I can remove them if you want.

**Comment by silverwind** [comment]
No, it's fine then. We could deprecate them later, but there's likely some reason behind them.

**Pull request merged**